File: app/src/main/java/wangdaye/com/geometricweather/db/converters/WindDegreeConverter.java
package wangdaye.com.geometricweather.db.converters;

import androidx.room.TypeConverter;

import wangdaye.com.geometricweather.common.basic.models.weather.WindDegree;

public class WindDegreeConverter {

    @TypeConverter
    public WindDegree convertToEntityProperty(Float databaseValue) {
        if (databaseValue == null) {
            return new WindDegree(-1, true);
        } else {
            return new WindDegree(databaseValue, false);
        }
    }

    @TypeConverter
    public Float convertToDatabaseValue(WindDegree entityProperty) {
        if (entityProperty == null || entityProperty.isNoDirection()) {
            return null;
        } else {
            return entityProperty.getDegree();
        }
    }
}

File: data/src/main/java/wangdaye/com/geometricweather/db/converters/WindDegreeConverter.kt
package wangdaye.com.geometricweather.db.converters

import androidx.room.TypeConverter
import wangdaye.com.geometricweather.common.basic.models.weather.WindDegree

class WindDegreeConverter {

    @TypeConverter
    fun convertToEntityProperty(databaseValue: Float?): WindDegree {
        return if (databaseValue == null) {
            WindDegree(-1f, true)
        } else {
            WindDegree(databaseValue, false)
        }
    }

    @TypeConverter
    fun convertToDatabaseValue(entityProperty: WindDegree?): Float? {
        return if (entityProperty == null || entityProperty.isNoDirection) {
            null
        } else {
            entityProperty.degree
        }
    }
}
